scripts/Interface/ui/reflectometer/refl_gui.py

```diff
@@ -1028,24 +1038,44 @@ def _do_run(self, runno, row, which):
+                                                                       OutputWorkspace=runno+'_IvsQ_'+str(i+1),
+                                                                       OutputWorkspaceWavelength=runno+'_IvsLam_'+str(i+1),
+                                                                       ScaleFactor=factor,MomentumTransferStep=Qstep,
+                                                                       MomentumTransferMin=Qmin, MomentumTransferMax=Qmax)
+
+                    wqGroupBinned.append(wqBinned)
                     wqGroup.append(wq)
                     wlamGroup.append(wlam)
                     thetaGroup.append(th)
 
+                wqBinned = GroupWorkspaces(InputWorkspaces=wqGroup, OutputWorkspace=runno+'_IvsQ_binned')
                 wq = GroupWorkspaces(InputWorkspaces=wqGroup, OutputWorkspace=runno+'_IvsQ')
                 wlam = GroupWorkspaces(InputWorkspaces=wlamGroup, OutputWorkspace=runno+'_IvsLam')
                 th = thetaGroup[0]
             else:
-                wq, wlam, th = ReflectometryReductionOneAuto(InputWorkspace=ws, FirstTransmissionRun=transmission_ws,
-                                                             thetaIn=angle, OutputWorkspace=runno+'_IvsQ',
-                                                             OutputWorkspaceWavelength=runno+'_IvsLam',
-                                                             ScaleFactor=factor,MomentumTransferStep=Qstep,
-                                                             MomentumTransferMinimum=Qmin, MomentumTransferMaximum=Qmax)
+                _wq, _wlam, th = ReflectometryReductionOneAuto(InputWorkspace=ws, FirstTransmissionRun=transmission_ws,
+                                                               thetaIn=angle, OutputWorkspace=runno+'_IvsQ',
+                                                               OutputWorkspaceWavelength=runno+'_IvsLam',
+                                                               ScaleFactor=factor,MomentumTransferStep=Qstep,
+                                                               MomentumTransferMinimum=Qmin, MomentumTransferMaximum=Qmax,
+                                                               Version=1)
+
+                wqBinned, wq, wlam = ReflectometryReductionOneAuto(InputWorkspace=ws, FirstTransmissionRun=transmission_ws,
+                                                                   thetaIn=angle, OutputWorkspaceBinned=runno+'_IvsQ_binned',
+                                                                   OutputWorkspace=runno+'_IvsQ',
+                                                                   OutputWorkspaceWavelength=runno+'_IvsLam',
+                                                                   ScaleFactor=factor,MomentumTransferStep=Qstep,
+                                                                   MomentumTransferMin=Qmin, MomentumTransferMax=Qmax)
 
```

Comment:
I think I came up with a better solution: If you replace lines `1066` to `1078` with:

```
                alg = AlgorithmManager.create("ReflectometryReductionOneAuto")
                alg.initialize()
                alg.setProperty("InputWorkspace", ws)
                alg.setProperty("FirstTransmissionRun", transmission_ws)
                if (angle != None):
                    alg.setProperty("ThetaIn", angle)
                alg.setProperty("OutputWorkspaceBinned", runno+'_IvsQ_binned')
                alg.setProperty("OutputWorkspace", runno+'_IvsQ')
                alg.setProperty("OutputWorkspaceWavelength", runno+'_IvsLam')
                alg.setProperty("ScaleFactor", factor)
                alg.setProperty("MomentumTransferStep", Qstep)
                alg.setProperty("MomentumTransferMin", Qmin)
                alg.setProperty("MomentumTransferMax", Qmax)
                alg.execute()
                wlam = mtd[runno+'_IvsLam']
                wq = mtd[runno+'_IvsQ']
                wqBinned = mtd[runno+'_IvsQ_binned']
                th = alg.getProperty("ThetaIn").value
```

we get the same functionality without having to run the algorithm twice. The same should work for lines `1042` - `1055`, but I haven't tried yet.